test/config/utility.h

```diff
@@ -143,9 +143,16 @@ class ConfigHelper {
   // Modifiers will be applied just before ports are modified in finalize
   void addConfigModifier(HttpModifierFunction function);
 
+  // Stick all the listeners in a discovery response message.
+  envoy::api::v2::DiscoveryResponse createLdsResponse(absl::string_view version_info) const;
+
   // Return the bootstrap configuration for hand-off to Envoy.
   const envoy::config::bootstrap::v2::Bootstrap& bootstrap() { return bootstrap_; }
 
+  // Allow a finalized configuration to be edited for generating xDS responses
+  void allowFurtherEdits() { finalized_ = false; }
```

Comment:
Can we do something less stateful here? I'm thinking specifically that it's easier to reason about an immutable bootstrap and then immutable updates to this. Why does `ConfigHelper` need to be edited, could we have a second `ConfigHelper` instantiated independently? Can we decompose `ConfigHelper` into parts that deal with downstream (LDS) and upstream (CDS)?